File: Java/src/main/java/Solution201_300/Solution268.java
package Solution201_300;

/**
 * @author windows
 */
public class Solution268 {
    public int missingNumber(int[] nums) {
        for (int i = 0; i < nums.length; i++) {
            while(nums[i] != nums.length && nums[i] != i) {
                int temp = nums[nums[i]];
                nums[nums[i]] = nums[i];
                nums[i] = temp;
            }
        }
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != i) {
                return i;
            }
        }
        return nums.length;
    }


    public static void main(String[] args) {
        Solution268 solution268 = new Solution268();
        System.out.println(solution268.missingNumber(new int[]{9,6,4,2,3,5,7,0,1}));
    }
}

File: C++/src/Solution201_300/Solution268.cpp
//
// Created by windows on 2023/3/1.
//

#include "Solution268.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;
#include <algorithm>

int Solution268::missingNumber(vector<int> &nums) {
    sort(nums.begin(), nums.end());
    for (int i = 0; i < nums.size(); i++) {
        if (nums[i] != i) {
            return i;
        }
    }
    return nums.size();
}
